packages/react-ui-core/src/Gmap/utils/mapEventHelpers.js

```diff
@@ -0,0 +1,13 @@
+export const handleEvent = (instance, name, props) =>
+  event => props[name](event, props, instance)
+
+export const setupEvents = (instance, events, props) => {
+  const eventNames = Object.keys(events)
+
+  eventNames.forEach(name => {
+    if (props[name] && typeof props[name] === 'function') {
+      const event = handleEvent(instance, name, props)
+      window.google.maps.event.addListener(instance, events[name], event)
```

Comment:
Is this assuming too much? (that `window.google.maps...` will be available?)

Reply:
I mean, if we don't have google maps, we have a problem.

However, yeh, might just want to make an isGoogleMapsAvailable checker or something so we don't throw errors

Reply:
Yeah, we can add a checker, but yeah, the way these work with the markers and such is that the api has to be loaded and the map has to be initialized before any of the children are mounted.